### PR TITLE
fix: allow diskInfo at storageRPC to be cached

### PR DESCRIPTION
**Pull request by harshavardhana**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
fix: allow diskInfo at storageRPC to be cached

## Motivation and Context
Bonus: convert timedValue into a typed implementation

## How to test this PR?
CI/CD should cover it, unit tests updated. 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Review by poornas — APPROVED** [review]

File: cmd/erasure-server-pool.go [hunk 1]
```diff
@@ -1001,20 +1001,10 @@ func (z *erasureServerPools) PutObject(ctx context.Context, bucket string, objec
...
-		if !isMinioMetaBucketName(bucket) {
-			avail, err := hasSpaceFor(getDiskInfos(ctx, z.serverPools[0].getHashedSet(object).getDisks()...), data.Size())
-			if err != nil {
-				logger.LogOnceIf(ctx, err, "erasure-write-quorum")
-				return ObjectInfo{}, toObjectErr(errErasureWriteQuorum)
-			}
-			if !avail {
-				return ObjectInfo{}, toObjectErr(errDiskFull)
-			}
-		}
```
Comment (vadmeste) [inline cmd/erasure-server-pool.go:1015]: Just wanted to check if this is intended. This does not seem in the scope of this PR.

**Review by vadmeste — COMMENTED** [review]
One comment

File: cmd/erasure-server-pool.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/erasure-server-pool.go:1015]: It is intended @vadmeste we do not have to do premature checks here.

Lower layers return appropriate errors properly when this happens.

We shouldn't really have to worry about drive space in this manner for every I/O it is quite expensive.

https://www.brendangregg.com/blog/2016-09-03/sudden-disk-busy.html

> All coming from stat(). The developers then realized they had enabled an app called "DiskUsageMonitor", which constantly stat()s the file system to track used space.

We shouldn't have to pay the price for something that will happen like 2yrs down the line.

**Review by vadmeste — APPROVED** [review]
LGTM

**Review by krisis — APPROVED** [review]

**Pull request merged**